src/lib/dnssd/minimal_mdns/AddressPolicy_DefaultImpl.cpp

```diff
@@ -22,103 +22,49 @@ namespace mdns {
 namespace Minimal {
 namespace {
 
-/// Checks if the current interface is powered on
-/// and not local loopback.
-template <typename T>
-bool IsCurrentInterfaceUsable(T & iterator)
+class SimpleInterfacesIterator : public mdns::Minimal::ListenIterator
 {
-    if (!iterator.IsUp())
-    {
-        return false; // not a usable interface
-    }
-    char name[chip::Inet::InterfaceId::kMaxIfNameLength];
-    if (iterator.GetInterfaceName(name, sizeof(name)) != CHIP_NO_ERROR)
-    {
-        ChipLogError(Discovery, "Failed to get interface name.");
-        return false;
-    }
-
-    // TODO: need a better way to ignore local loopback interfaces/addresses
-    // We do not want to listen on local loopback even though they are up and
-    // support multicast
-    //
-    // Some way to detect 'is local looback' that is smarter (e.g. at least
-    // strict string compare on linux instead of substring) would be better.
-    //
-    // This would reject likely valid interfaces like 'lollipop' or 'lostinspace'
-    if (strncmp(name, "lo", 2) == 0)
+private:
+    enum class ReturnState
     {
-        /// local loopback interface is not usable by MDNS
-        return false;
-    }
-    return true;
-}
-
-class AllInterfaces : public mdns::Minimal::ListenIterator
-{
+        kStart,
+#if INET_CONFIG_ENABLE_IPV4
+        kIPv4,
+#endif
+        kEnd,
+    };
 public:
-    AllInterfaces() { SkipToFirstValidInterface(); }
+    SimpleInterfacesIterator() : mState(ReturnState::kStart) {}
 
     bool Next(chip::Inet::InterfaceId * id, chip::Inet::IPAddressType * type) override
     {
-        if (!mIterator.HasCurrent())
-        {
-            return false;
-        }
+        *id = chip::Inet::InterfaceId::Null();
```

Comment:
This will skip several interfaces, which may be unrelated. If a node is reachable operationally at a given interface, then we should mDNS listen on it as well.

Reply:
This listens on all interfaces ... there is no "skip"

Reply:
Or I guess viewed another way, this skipps trying to figure out interfaces as it relies a "listen on none" to be equivalent to listen on all possible interfaces. This makes the code use a single (or 2 when ipv4 is enabled) udp endpoint for all tasks.

With this PR I am claiming that the idea of binding to every interface individually was probably wrong. I believe it was done because SendTo by default would not work over link local addresses, but I solved that by using the interface ID argument (not sure if that argument existed when minmdns code was first written).